app/actions/instancesManager.js

```diff
@@ -41,10 +41,18 @@ export function startInstance(instanceName) {
   return async (dispatch, getState) => {
     const { auth, settings } = getState();
 
+    const config = JSON.parse(
+      await promisify(fs.readFile)(
+        path.join(PACKS_PATH, instanceName, 'config.json')
+      )
+    );
+
     const command = await launchCommand(
       instanceName,
       auth,
-      settings.java.memory,
+      config.overrideMemory === ''
```

Comment:
it should be`config.overrideMemory === undefined` since when no override is set, it is simply not present